<launch>
	
	<!-- below are all the Parameters defined that are needed to run three Syringes!!! -->

	<arg name="pump1" default="medium_tank" doc=""/>
	<arg name="step_size1" default="110" doc=""/>
	<arg name="pins1" default="13,15,19,21" doc=""/>
	<arg name="node_name1" default="aroma_syringe_node1" doc=""/>

	<arg name="pump2" default="spill_tank" doc=""/>
	<arg name="step_size2" default="110" doc=""/>
	<arg name="pins2" default="12,16,18,22" doc=""/>
	<arg name="node_name2" default="aroma_syringe_node2" doc=""/>

	<arg name="pump3" default="microfluidic" doc=""/>
	<arg name="step_size3" default="3650" doc=""/>
	<arg name="pins3" default="31,29,26,24" doc=""/>
	<arg name="node_name3" default="aroma_syringe_node3" doc=""/>


  <!-- below are all the Parameters defined that are needed for the driving!!! -->

  <arg name="pins4" default="37,35,23,33,36,32,40,38" doc="Specify the pins your wheels are connected to"/>
  <arg name="node_name4" default="aroma_drive_node" doc="Name of the node"/>
  <arg name="straight_speed1" default="0.328" doc="Is a relative factor to adapt the speed in m/s. Has to be properly calibrated"/>
  <arg name="angle_speed1" default="48.8" doc="Is a relative factor that defines the turning speed in deg/s. Has to be properly calibrated!"/>


  <!-- below are all the Parameters defined that are needed for the airpump!!! -->

  <arg name="pins5" default="7" doc="Specify the pins your wheels are connected to"/>
  <arg name="node_name5" default="aroma_airpump_node" doc="Name of the node"/>


  <!-- below are all the Parameters that are needed for the FSM itself,...!!! -->



 <!-- below are all the Parameters that are needed for the Finite State Machine itself!!! -->

  <param name="~pump1" value="$(arg pump1)"/>
  <param name="~pump2" value="$(arg pump2)"/>
  <param name="~pump3" value="$(arg pump3)"/>

  <node name="aroma_fsm_node" pkg="aroma_fsm" type="fsm.py" output="screen">
    </node>


  <!-- below happen all the Launches!!! -->
    
    <include file="$(find aroma_syringe)/launch/aroma_syringe.launch">
      <arg name="name" value="$(arg pump1)" />
      <arg name="step_size" value="$(arg step_size1)" />
      <arg name="pins" value="$(arg pins1)" />
      <arg name="node_name" value="$(arg node_name1)" />

    </include>

  	<include file="$(find aroma_syringe)/launch/aroma_syringe.launch">
      <arg name="name" value="$(arg pump2)" />
      <arg name="step_size" value="$(arg step_size2)" />
      <arg name="pins" value="$(arg pins2)" />
      <arg name="node_name" value="$(arg node_name2)" />

    </include>

   	<include file="$(find aroma_syringe)/launch/aroma_syringe.launch">
      <arg name="name" value="$(arg pump3)" />
      <arg name="step_size" value="$(arg step_size3)" />
      <arg name="pins" value="$(arg pins3)" />
      <arg name="node_name" value="$(arg node_name3)" />

    </include>

    <include file="$(find aroma_drive)/launch/aroma_drive.launch">
      <arg name="pins" value="$(arg pins4)" />
      <arg name="node_name" value="$(arg node_name4)" />
      <arg name="straight_speed" value="$(arg straight_speed1)" />
      <arg name="angle_speed" value="$(arg angle_speed1)" />

    </include>

    <include file="$(find aroma_airpump)/launch/aroma_airpump.launch">
      <arg name="pins" value="$(arg pins5)" />
      <arg name="node_name" value="$(arg node_name5)" />

    </include>


</launch>
